src/lib.rs

```diff
@@ -1,165 +1,174 @@
-// File operations used for modifying LED state
 use std::fs::File;
 use std::io;
 use std::io::Write;
 
-// Paths to spi daemon sockets with incoming data from coprocessor
+// Paths to the SPI daemon sockets with incoming data from coprocessor.
 const PORT_A_UDS_PATH: &'static str = "/var/run/tessel/port_a";
 const PORT_B_UDS_PATH: &'static str = "/var/run/tessel/port_b";
 
-// Primary exported Tessel object with access to module ports, leds, and a button
-#[allow(dead_code)]
+/// Primary exported Tessel object with access to module ports, LEDs, and a button.
+/// # Example
+/// ```
+/// use rust_tessel::Tessel;
+///
+/// # #[allow(dead_code)]
+/// # fn example() {
+/// let t = Tessel::new();
+/// // Tessel 2 has four LEDs available.
+/// assert_eq!(t.led.len(), 4);
+/// // Tessel 2 has two ports labelled a and b
+/// let a = t.port.a;
+/// let b = t.port.b;
+/// // Tessel 2 has one button.
+/// let button = t.button;
+/// # }
+/// ```
 pub struct Tessel {
-    // A group of module ports
+    // A group of module ports.
     pub port: PortGroup,
-    // An array of LED structs
+    // An array of LED structs.
     pub led: Vec<LED>,
-    // A single button struct
-    pub button: Button
+    // A single button struct.
+    pub button: Button,
 }
 
 impl Tessel {
-    // Factory metho for Tessel
     pub fn new() -> Tessel {
 
-        // Create a port group with two ports, one on each domain socket path
-        let ports = PortGroup { a: Port::new(PORT_A_UDS_PATH), b: Port::new(PORT_B_UDS_PATH) };
+        // Create a port group with two ports, one on each domain socket path.
+        let ports = PortGroup {
+            a: Port { socket_path: PORT_A_UDS_PATH },
+            b: Port { socket_path: PORT_B_UDS_PATH },
+        };
 
-        // Create models for the four LEDs
+        // Create models for the four LEDs.
         let red_led = LED::new("red", "error");
         let amber_led = LED::new("amber", "wlan");
         let green_led = LED::new("green", "user1");
         let blue_led = LED::new("blue", "user2");
 
-        // Create the button
-        let button =  Button::new();
+        // Create the button.
+        let button = Button { value: false };
```

Comment:
Per https://github.com/tessel/t2-firmware/issues/107, we don't want to expose the button in the API after all.